### ISSUE TITLE
Alan does not recognize/record voice input on specific Android devices

### ISSUE THREAD

**Issue by ljudbane** [alan-ai/alan-sdk-ionic#25]
We have encountered an issue where on specific Android devices the voice input is not recognized. When Alan is listening, nothing happens, as if the voice is not recorded. No amount of app re-installs or different builds have helped, the issue is always present.

Affected devices are:
- Samsung Galaxy S20 FE (SM-G780F/DS), OneUI 3.1, Android 11, sec patch 1.nov.2021
- Samsung Galaxy Note 10 (SM-N970F/DS), OneUI 3.1, Android 11

On Samsung S20 FE we tested your reference app "Alan Playground" (installed from Google Play) and voice input also doesn't work in your app. We will also test your app on Note 10 device at a later time and we will update the issue once we get results.

I am attaching logcat from running our Ionic app on Samsung S20 FE. In the log i've marked the approximate location when user started speaking (line 416). A bit above that, on line 398 is app's log entry when button state callback fired and state was LISTEN. So somewhere around here Alan started listening.

[samsung_s20_fe_dg1_app_logcat-edit.log](https://github.com/alan-ai/alan-sdk-ionic/files/7704476/samsung_s20_fe_dg1_app_logcat-edit.log)

Ionic info:
```
Ionic:

   Ionic CLI                     : 5.4.16 
   Ionic Framework               : @ionic/angular 5.3.2
   @angular-devkit/build-angular : 0.803.29
   @angular-devkit/schematics    : 8.3.29
   @angular/cli                  : 8.3.29
   @ionic/angular-toolkit        : 2.3.3

Capacitor:

   Capacitor CLI   : 2.4.0
   @capacitor/core : 2.4.0

Utility:

   cordova-res                          : not installed
   native-run (update available: 1.5.0) : 1.3.0

System:

   NodeJS : v10.18.1
   npm    : 6.13.4
   OS     : macOS Big Sur
```

**Comment by snyuryev** [comment]
Hey @ljudbane 
Thank you for detailed issue. We will try to reproduce it.

**Comment by ljudbane** [comment]
@snyuryev Yes, the permission popup was shown and approved. We also double checked in Android settings and permission for recording audio was enabled in App settings. The microphone does work with phone calls and other apps.
